src/workloads/query/MatchWithLargeExpression.yml

```diff
@@ -0,0 +1,150 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  This workload tests the performance of expression search for parameter re-use during parameterization. 
+  Before September 2023, parameter re-use had O(n^2) complexity due to using a vector for looking up equivalent expressions. 
+  SERVER-79092 fixed this issue by switching over to a map once the amount of expressions reaches a threshold (currently 50).
```

Comment:
Are we planning to commit this before SERVER-79092 so that we can see the improvement? Although it looks like you already have this in the PR for SERVER-79092.

Reply:
What metrics are important for this workload, throughput, latency, .... etc?

Reply:
Probably throughput is more important, but this change will lower the latency per-query from my understanding- why not both?

Reply:
> I was thinking of committing this first but can do it either way. Let me know if you have a preference.

Committing first is great! If you take this approach then you should mention on the perf-build-baroning channel that an improvement is expected when the SERVER ticket is committed.
  
> Both operation throughput and latency should be and are improved greatly by SERVER-79092. If one of them isn't, something might be wrong. So I'd say they are both important.

Cool, can you mention in the description that these are the important metrics for this workload?